safe_transaction_service/history/views.py

```diff
@@ -822,6 +859,88 @@ def delete(self, request, delegate_address, *args, **kwargs):
             return Response(status=status.HTTP_404_NOT_FOUND)
 
 
+class TransferView(GenericAPIView):
+    serializer_class = serializers.TransferWithTokenInfoResponseSerializer
+    pagination_class = None
+
+    def get_erc20_erc721_transfer(self, transaction_hash: str, log_index: str):
+        """
+        Search ERCTransfer by transaction_hash and log_index event.
+
+        :param transaction_hash: ethereum transaction hash
+        :param log_index: event log index
+        :return: ERC20Transfer object
+        """
+        erc20_queryset = self.filter_queryset(
+            ERC20Transfer.objects.filter(
+                ethereum_tx=transaction_hash, log_index=log_index
+            ).token_txs()
+        )
+        erc721_queryset = self.filter_queryset(
+            ERC721Transfer.objects.filter(
+                ethereum_tx=transaction_hash, log_index=log_index
+            ).token_txs()
+        )
+        return ERC20Transfer.objects.token_transfer_values(
+            erc20_queryset, erc721_queryset
+        )
+
+    def get_ethereum_transfer(self, transaction_hash: str, trace_address: str):
+        """
+        Search an ethereum transfer by transaction hash and trace address
+
+        :param transaction_hash: ethereum transaction hash
+        :param trace_address: ethereum trace address
+        :return: InternalTx object
+        """
+        ether_queryset = self.filter_queryset(
+            InternalTx.objects.ether_txs().filter(
+                ethereum_tx=transaction_hash, trace_address=trace_address
+            )
+        )
+        return InternalTx.objects.ether_txs_values(ether_queryset)
+
+    def get_queryset(self, transfer_id):
+        # transfer_id is composed by transfer_type (ethereum transfer or token_transfer) + tx_hash + (log_index or trace_address)
+        transfer_type = transfer_id[0]
+        tx_hash = transfer_id[1:65]
+        log_or_trace = transfer_id[65:]
+        if transfer_type == "i":
+            # It is an ethereumTransfer
+            return self.get_ethereum_transfer(tx_hash, log_or_trace)
+        else:
+            # It is an tokenTransfer
+            return self.get_erc20_erc721_transfer(tx_hash, log_or_trace)
+
+    def add_tokens_to_transfer(self, transfer: TransferDict) -> TransferDict:
+        tokens = {
+            token.address: token
+            for token in Token.objects.filter(address__in={transfer["token_address"]})
+        }
+        transfer["token"] = tokens.get(transfer["token_address"])
+
+        return transfer
+
+    def get(self, request, transfer_id, *args, **kwargs):
+        """
+        Return a transfer filtered by transfer_id
```

Comment:
Use RST for comments
![image](https://user-images.githubusercontent.com/6909403/236201818-2d08c6b6-dc84-4e64-bbcb-024d33dcc2ae.png)

        
`:return: Transfer filtered by transfer_id`

Also, add types for the parameters